config/oauth2.go

```diff
@@ -29,7 +32,7 @@ func (oa OAuth2) HCLBody() hcl.Body {
 
 // Reference implements the <Inline> interface.
 func (oa OAuth2) Reference() string {
-	return "OAuth2"
+	return oa.BackendName
```

Comment:
`if oa.BackendName == "" { return "OAuth2" } else { return oa.BackendName }`?

Reply:
Since this would be for logging purposes only the current `token_request` log field should be ok for now.